go/nbs/table_reader.go

```diff
@@ -38,21 +40,28 @@ func newTableReader(buff []byte, r io.ReaderAt, readAmpThresh uint64) tableReade
 	pos -= uint64Size
 
 	pos -= uint32Size
-	tr.chunkCount = binary.BigEndian.Uint32(buff[pos:])
+	chunkCount := binary.BigEndian.Uint32(buff[pos:])
 
 	// index
-	suffixesSize := uint64(tr.chunkCount) * addrSuffixSize
+	suffixesSize := uint64(chunkCount) * addrSuffixSize
 	pos -= suffixesSize
-	tr.suffixes = buff[pos : pos+suffixesSize]
+	suffixes := make([]byte, suffixesSize)
+	copy(suffixes, buff[pos:])
```

Comment:
Per offline conversation, it seems that the goal is to make the table index a wrapper around a memory-mapped piece of a file no matter what the ultimate storage backend is. That sounds fine to me.